### PR TITLE
Restore port 8081 usage for new minor version

### PR DESCRIPTION
**Pull request by pega-talba**

### CODE REVIEW

**Review by APegaDavis — DISMISSED** [review]

**Comment by petejo** [comment]
To avoid a dependency on the docker-pega-web-ready image enhancement providing server.xml injection, let's explicitly add the various probe URLs to the values.yaml, and default to port 8080 in the helm templates when not provided (e.g. existing  values.yaml)

File: README.md [hunk 1]
```diff
@@ -160,3 +160,15 @@ These images do not expire, and you can keep them in your repository for as long
 # Contributing
 
 This is an open source project and contributions are welcome.  Please see the [contributing guidelines](./CONTRIBUTING.md) to get started.
+
+# Versioning
```
Comment (taz-pega-work) [inline README.md:164]: # Versioning of the Pega Helm repository

Pega may release new versions of this Pega Helm chart released at any time. (Reviewers: do you want to add something about givving clients a heads up about future versions or is it not required?) Versions are defined as major.minor.patch, as follows:

1. Major: Used when required values.yaml or the config maps (config/deploy) do not adequately support generating Kubernetes yaml and deploy using previously released Docker image versions.
  - Example: A dependency on a specific Pega Platform version, or a dependency on either a required pega-web-ready or a pega-installer-ready image change.
2. Minor: Used when the older values.yaml or the config maps (config/deploy) no longer support the Helm parameters to adopt new functionality and to generate the Kubernetes yaml. Minor may also be used for new features.
  - Example: A dependency on an optional pega-web-ready change or pega-installer-ready image change.
3. Patch: Any other change, such opt-in enhancements  and bugs fixes, that Pega deems important enough to address between minor releases.
  - Example: Introducing a new, optional parameter that defaults to a previously hard-coded value.

**Review by taz-pega-work — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pega-talba) [reply README.md:164]: After more discussion with John and others, we've decided to take on a more standard versioning approach (by just using semantic versioning exactly as it is).

File: README.md [hunk 2]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
-  - Example: A dependency on a specific Platform Version, or a dependency on either a required pega-web-ready or a pega-installer-ready change.
-2. Minor: Used when the older vaules.yaml or the config maps (config/deploy) cannot be provided as input to Helm for generating the K8S yaml without being updated to adopt the new functionality. Minor may also be used for new features.
-  - Example: A dependency on an optional pega-web-ready change or pega-installer-ready change.
-3. Patch: Any other change, such that enhancements are opt-in and bugs are fixed to behave in the expected manner.
-  - Example: A new optional parameter that defaults to a previously hard-coded value
+New versions of this Helm Chart may be released at any time. Versions are defined using [semantic versioning](https://semver.org/):
+
+1. Major: Breaks backwards compatibility with previous chart versions. Examples:
+  a. The new values.yaml or maps (config/deploy) cannot be used to deploy previously released Docker image versions
```
Comment (taz-pega-work) [inline README.md:169]: Make all of the a. & b. bullets, not substeps. These bullets need periods at the end of all of the sentences

File: README.md [hunk 3]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+  b. There is a new, required dependency on a specific Platform or dependent docker image
```
Comment (taz-pega-work) [inline README.md:170]: specific Pega Platform version

File: README.md [hunk 4]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+2. Minor: Adds new functionality and maintains backwards compatibility. Examples:
+  a. A new feature is added but existing functionality is maintained
```
Comment (taz-pega-work) [inline README.md:172]: Pega adds new features while maintaining existing functionality.

File: README.md [hunk 5]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+  b. An opt-in configuration change is made
```
Comment (taz-pega-work) [inline README.md:173]: Support for new, opt-in configurations.

File: README.md [hunk 6]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
-
-1. Major: Used when there is no way to configure the values.yaml or the config maps (config/deploy) such that the generated K8S yaml can be used to deploy previously released Docker image versions.
-  - Example: A dependency on a specific Platform Version, or a dependency on either a required pega-web-ready or a pega-installer-ready change.
-2. Minor: Used when the older vaules.yaml or the config maps (config/deploy) cannot be provided as input to Helm for generating the K8S yaml without being updated to adopt the new functionality. Minor may also be used for new features.
-  - Example: A dependency on an optional pega-web-ready change or pega-installer-ready change.
-3. Patch: Any other change, such that enhancements are opt-in and bugs are fixed to behave in the expected manner.
-  - Example: A new optional parameter that defaults to a previously hard-coded value
+New versions of this Helm Chart may be released at any time. Versions are defined using [semantic versioning](https://semver.org/):
+
+1. Major: Breaks backwards compatibility with previous chart versions. Examples:
```
Comment (taz-pega-work) [inline README.md:168]: Major: Pega introduces new features or functionality that results in breaking backwards compatibility with previous chart versions. Examples:

File: README.md [hunk 7]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+  a. The new values.yaml or maps (config/deploy) cannot be used to deploy previously released Docker image versions
+  b. There is a new, required dependency on a specific Platform or dependent docker image
+2. Minor: Adds new functionality and maintains backwards compatibility. Examples:
+  a. A new feature is added but existing functionality is maintained
+  b. An opt-in configuration change is made
+3. Patch: Fixes bugs and maintains backwards compatibility. Examples:
```
Comment (taz-pega-work) [inline README.md:174]: Patch: Pega fixes bugs and maintains backwards compatibility. Examples:

File: README.md [hunk 8]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+  a. A bug or issue is fixed
```
Comment (taz-pega-work) [inline README.md:175]: Pega fixes a bug or addresses known issues.

File: README.md [hunk 9]
```diff
@@ -163,12 +163,15 @@ This is an open source project and contributions are welcome.  Please see the [c
...
+  b. A security vulnerability is fixed
```
Comment (taz-pega-work) [inline README.md:176]: Pega addresses a security vulnerability.

**Review by taz-pega-work — APPROVED** [review]

**Review by taz-pega-work — APPROVED** [review]
where are these different probe types documented?
the helm charts exclude startupProbe and use different default values for livenessProbe and readinessProbe.

I don't have enough context...just be sure those various probes are clearly explained. I thought I did a different review with those details...leaving it up to you. This new content is great!

**Comment by pega-talba** [comment]
> 
> 
> where are these different probe types documented?
> the helm charts exclude startupProbe and use different default values for livenessProbe and readinessProbe.
> 
> I don't have enough context...just be sure those various probes are clearly explained. I thought I did a different review with those details...leaving it up to you. This new content is great!

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
They are documented by Kubernetes here ^ and that link is also included in the README where we discuss the probes

**Review by APegaDavis — APPROVED** [review]

**Pull request merged**